dlt/destinations/sql_jobs.py

```diff
@@ -253,20 +292,36 @@ def gen_merge_sql(
         sql: List[str] = []
         root_table = table_chain[0]
 
+        escape_identifier = sql_client.capabilities.escape_identifier
+        escape_literal = sql_client.capabilities.escape_literal
+
+        insert_condition = "1 = 1"
+        write_disposition = root_table["write_disposition"]
+        if write_disposition == "replicate":
```

Comment:
IMO SqlMergeJob should not be aware of write disposition... Is this possible to create a base class and two subclasses for merge and replicate.

this is very hacky